src/PollySandbox/ResilienceStrategyStore.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Martin Costello, 2023. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.Net;
+using System.Threading.RateLimiting;
+using Microsoft.Extensions.Caching.Memory;
+using Polly;
+using Polly.CircuitBreaker;
+using Polly.RateLimiting;
+using Polly.Retry;
+using Polly.Timeout;
+using Refit;
+
+namespace PollySandbox;
+
+public partial class ResilienceStrategyStore
+{
+    private readonly IMemoryCache _cache;
+    private readonly ILogger _logger;
+    private readonly ConcurrentDictionary<string, ResilienceStrategy> _registry;
+    private readonly IMetricsPublisher _metricsPublisher;
+    private readonly ConcurrentDictionary<string, CircuitBreakerManualControl> _isolators;
```

Comment:
Introduce `CircuitBreakerIsolators` into DI so it can be used when configuring the strategy using `AddResilienceStrategy` extension?

Reply:
Yeah, I'm playing around with using the DI extensions to replace the custom classes now. I'm sure it'll be fine once it all comes together, but the state the current class has (e.g. for the loggers and the metrics) are making it a bit tricker to port over.